qtawesome/icon_browser.py

```diff
@@ -270,7 +270,14 @@ def run():
     browser = IconBrowser()
     browser.show()
 
-    sys.exit(app.exec_())
+    # exec_ function is deprecated in PySide6.
+    # Getting exec/exec_ and executing it is separated
+    # because executing exec/exec_ may propagates any exceptions.
+    try:
+        app_exec = app.exec
+    except AttributeError:
+        app_exec = app.exec_
+    sys.exit(app_exec())
```

Comment:
I was thinking about this and maybe is not needed? Or in what case `exec` in not available?

Reply:
> Or in what case exec in not available?

For example, when QT_API=pyside2, QtWidgets.QApplication does not have `exec`. It have `exec_` only.
As https://github.com/spyder-ide/qtpy/issues/286, if QtPy will provide `exec_` of PySide6 which does not report DeprecationWarning, this change is not required.

Reply:
Thanks for the clarification @kumattau ! Then I think we should keep here the call for `exec_` and as you say provide over QtPy the aliasing/mapping from `exec` to `exec_` for PySide6 to prevent the DeprecationWarning :+1: